legend-engine-application-query/src/main/java/org/finos/legend/engine/application/query/api/QueryStoreManager.java

```diff
@@ -386,13 +387,50 @@ else if (matchingQueries.size() == 0)
         query.owner = currentUser;
         query.createdAt = currentQuery.createdAt;
         query.lastUpdatedAt = Instant.now().toEpochMilli();
+        query.originalVersionId = currentQuery.originalVersionId;
         this.getQueryCollection().findOneAndReplace(Filters.eq("id", queryId), queryToDocument(query));
         QueryEvent updatedEvent = createEvent(query.id, QueryEvent.QueryEventType.UPDATED);
         updatedEvent.timestamp = query.lastUpdatedAt;
         this.getQueryEventCollection().insertOne(queryEventToDocument(updatedEvent));
         return query;
     }
 
+    public Query updateQuerySelectedFields(String queryId, Query updatedQuery, String currentUser) throws JsonProcessingException
+    {
+        Query currentQuery = this.getQuery(queryId);
+        // Make sure only the owner can update the query
+        // NOTE: if the query is created by an anonymous user previously, set the current user as the owner
+        if (currentQuery.owner != null && !currentQuery.owner.equals(currentUser))
+        {
+            throw new ApplicationQueryException("Only owner can update the query", Response.Status.FORBIDDEN);
+        }
+
+        Class<? extends Query> queryClazz = currentQuery.getClass();
```

Comment:
Is this a typo?
```suggestion
        Class<? extends Query> queryClass = currentQuery.getClass();
```

Reply:
👀 I think it's just a naming convention per each developer's preference loll 👀😅😅

Reply:
Oh interesting. Feels weird to me haha, but if it's used elsewhere then should be fine

Reply:
not a big deal but I think this modue is off in naming convention, we should probably start being more reader friendly and start naming things from clazz to class.